jwst/pipeline/calwebb_spec2.py

```diff
@@ -43,179 +48,226 @@ class Spec2Pipeline(Pipeline):
+                'NRS_MSASPEC', 'NIS_WFSS', 'NRC_GRISM'
+        ]:
 
-                # Save the cube product
-                if self.cube_build.skip != True:
-                    self.save_model(cube, 's3d')
-                    log.info('Saved IFU cube product to %s' % cube.meta.filename)
+            # Call the resample_spec step
+            resamp = self.resample_spec(input)
 
-                # Pass the cube along for input to 1D extraction
-                x1d_input = cube.copy()
-                cube.close()
+            # Save the resampled product
+            if not self.resample_spec.skip:
+                self.save_model(resamp, 's2d')
+                log.info(
+                    'Saved resampled product to %s' % resamp.meta.filename
+                )
 
-            else:
-                # Pass the unresampled cal product to 1D extraction
-                x1d_input = input
+            # Pass the resampled data to 1D extraction
+            x1d_input = resamp.copy()
+            resamp.close()
 
-            # Extract a 1D spectrum from the 2D/3D data
-            x1d_output = self.extract_1d(x1d_input)
-            x1d_input.close()
+        elif exp_type in ['MIR_MRS', 'NRS_IFU']:
 
-            # Save the extracted spectrum
-            if self.extract_1d.skip != True:
-                if isinstance(input, datamodels.CubeModel):
-                    self.save_model(x1d_output, 'x1dints')
-                else:
-                    self.save_model(x1d_output, 'x1d')
-                log.info('Saved extracted spectrum to %s' % x1d_output.meta.filename)
+            # Call the cube_build step for IFU data
+            cube = self.cube_build(input)
 
-            input.close()
-            x1d_output.close()
+            # Save the cube product
+            if not self.cube_build.skip:
+                self.save_model(cube, 's3d')
+                log.info('Saved IFU cube product to %s' % cube.meta.filename)
 
-        # We're done
-        log.info('Ending calwebb_spec2')
+            # Pass the cube along for input to 1D extraction
+            x1d_input = cube.copy()
+            cube.close()
+
+        else:
+            # Pass the unresampled cal product to 1D extraction
+            x1d_input = input
 
+        # Extract a 1D spectrum from the 2D/3D data
+        x1d_output = self.extract_1d(x1d_input)
+        x1d_input.close()
+
+        # Save the extracted spectrum
+        if not self.extract_1d.skip:
+            if isinstance(input, datamodels.CubeModel):
+                self.save_model(x1d_output, 'x1dints')
+            else:
+                self.save_model(x1d_output, 'x1d')
+            log.info(
+                'Saved extracted spectrum to %s' % x1d_output.meta.filename
+            )
+
+        input.close()
+        x1d_output.close()
+
+        # That's all folks
+        log.info(
+            'Finished processing exposure {}'.format(exp_product['name'])
```

Comment:
Same comment as earlier: perhaps this should say "Finished processing **product** ..."